src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/CreateFileFromTemplateService.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using EnvDTE;
+using EnvDTE80;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    /// <summary>
+    /// This service creates a file from a given file template.
+    /// </summary>
+    [Export(typeof(ICreateFileFromTemplateService))]
+    internal class CreateFileFromTemplateService : ICreateFileFromTemplateService
+    {
+        private IUnconfiguredProjectVsServices _projectVsServices;
+
+        [ImportingConstructor]
+        public CreateFileFromTemplateService(IUnconfiguredProjectVsServices projectVsServices)
+        {
+            Requires.NotNull(projectVsServices, nameof(projectVsServices));
+
+            _projectVsServices = projectVsServices;
+        }
+
+        /// <summary>
+        /// Get the language string to pass to the VS APIs for getting a template.
+        /// </summary>
+        private string GetTemplateLanguage(Project project)
+        {
+            switch (project.CodeModel.Language)
```

Comment:
Our project type already sets this the registry via "Language" key - long term we'll want to avoid hard-coding this so that other project types can opt-into our special files and have it *just* work.